python/lsst/obs/base/ingest.py

```diff
@@ -367,7 +367,16 @@ def extractMetadata(self, filename: ButlerURI) -> RawFileData:
                 with filename.as_local() as local_file:
                     phdu = readMetadata(local_file.ospath, 0)
                     header = merge_headers([phdu, readMetadata(local_file.ospath)], mode="overwrite")
```

Comment:
I have been wondering about always doing this second read. I have to do it because compressed files have a small primary header and a large secondary one. Phosim also puts a small amount of information in the primary but all the important information in the second.

One thing I could try is:

* Read primary header
* Ask whether translator knows what data this is.
* If YES: call read_all_headers
* If NO: read the second header (if there) and ask for the translator class again.

This might be slightly more efficient if the read_all_headers is going to read the second header anyhow.

Reply:
I implemented this and it seems to work okay.